clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchFieldMapper.scala

```diff
@@ -0,0 +1,99 @@
+package org.broadinstitute.clio.server.dataaccess.elasticsearch
+
+import java.net.URI
+import java.time.OffsetDateTime
+import java.util.UUID
+
+import com.sksamuel.elastic4s.mappings.FieldDefinition
+import enumeratum.EnumEntry
+import enumeratum.values.{IntEnum, IntEnumEntry}
+import org.broadinstitute.clio.server.dataaccess.elasticsearch.ElasticsearchIndex.TextExactMatchFieldName
+import org.broadinstitute.clio.util.model.UpsertId
+
+import scala.collection.immutable
+import scala.reflect.runtime.universe
+import scala.reflect.runtime.universe.Type
+
+private[dataaccess] sealed abstract class ElasticsearchFieldMapper(
+  override val value: Int
+) extends IntEnumEntry {
+
+  def stringToDefinition(fieldType: Type): (String => FieldDefinition)
+}
+
+private[dataaccess] object ElasticsearchFieldMapper
+    extends IntEnum[ElasticsearchFieldMapper] {
+  import com.sksamuel.elastic4s.http.ElasticDsl._
+
+  override val values: immutable.IndexedSeq[ElasticsearchFieldMapper] = findValues
+
+  /** All non-numeric, non-date members mapped to keywords. */
+  case object InitVersion extends ElasticsearchFieldMapper(1) {
```

Comment:
Having `Version` in the name at all is an accident from earlier logic... will clean up.

I'll also rename `Init` to be more descriptive about how it actually maps things. I didn't want to go with a `VersionN` pattern because it's not really more informative than just having an int. Not sure what you mean about the third version... if we end up needing it we'll have to make a new `case object` either way, and we can name that object to describe the change in mapping logic it introduces.